### PR TITLE
PACA-645: Handle CR outcome mappings

### PR DESCRIPTION
**Pull request by KyRobbins**
All outcomes need to be properly mapped from their initial FISS/MCS status codes, even for unknown status codes.

### What Should Reviewers Watch For?

If you're reviewing this PR, please check for these things in particular:

* Verify all PR security questions and checklists have been completed and addressed.

### What Security Implications Does This PR Have?

Submitters should complete the following questionnaire:

* If the answer to any of the questions below is **Yes**, then **you must** supply a link to the associated Security Impact Assessment (SIA), security checklist, or other similar document in Confluence here: **N/A**

    * Does this PR add any new software dependencies? 
      * [ ] Yes
      * [x] No
    * Does this PR modify or invalidate any of our security controls?
      * [ ] Yes
      * [x] No
    * Does this PR store or transmit data that was not stored or transmitted before?
      * [ ] Yes
      * [x] No

* If the answer to any of the questions below is **Yes**, then please add @StewGoin as a reviewer, and note that this PR **should not be merged** unless/until he also approves it.
    * Do you think this PR requires additional review of its security implications for other reasons?
      * [ ] Yes
      * [x] No

### What Needs to Be Merged and Deployed Before this PR?

This PR cannot be either merged or deployed until the following prerequisite changes have been fully deployed:

* N/A

### Submitter Checklist

I have gone through and verified that...:

* [x] I have named this PR and branch so they are [automatically linked](https://confluence.atlassian.com/adminjiracloud/integrating-with-development-tools-776636216.html) to the (most) relevant Jira issue. Ie: `BFD-123: Adds foo`
* [x] This PR is reasonably limited in scope, to help ensure that:
    1. It doesn't unnecessarily tie a bunch of disparate features, fixes, refactorings, etc. together.
    2. There isn't too much of a burden on reviewers.
    3. Any problems it causes have a small "blast radius".
    4. It'll be easier to rollback if that becomes necessary.
* [x] This PR includes any required documentation changes, including `README` updates and changelog / release notes entries.
* [x] All new and modified code is appropriately commented, such that the what and why of its design would be reasonably clear to engineers, preferably ones unfamiliar with the project.
* [x] All tech debt and/or shortcomings introduced by this PR are detailed in `TODO` and/or `FIXME` comments, which include a JIRA ticket ID for any items that require urgent attention.
* [x] Reviews are requested from both:
    * At least two other engineers on this project, at least one of whom is a senior engineer or owns the relevant component(s) here.
    * Any relevant engineers on other projects (e.g. DC GEO, BB2, etc.).
* [x] Any deviations from the other policies in the [DASG Engineering Standards](https://github.com/CMSgov/cms-oeda-dasg/blob/master/policies/engineering_standards.md) are specifically called out in this PR, above.
    * Please review the standards every few months to ensure you're familiar with them.

### CODE REVIEW

**Review by brianburton — DISMISSED** [review]

File: apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/pac/FissClaimResponseTransformerV2.java [hunk 1]
```diff
@@ -153,6 +173,22 @@ private static List<Identifier> getIdentifier(RdaFissClaim claimGroup) {
             .setValue(claimGroup.getDcn()));
   }
 
+  /**
+   * Maps the given status code to an associated {@link ClaimResponse.RemittanceOutcome}.
```
Comment (ericbuckley) [inline apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/pac/FissClaimResponseTransformerV2.java:177]:
```suggestion
   * Maps the given status code to an associated {@link ClaimResponse.RemittanceOutcome}. For unknown
   * status codes (aka codes not documented in the MCS data dictionary), we will default to the PARTIAL
   * outcome as the other 3 values (COMPLETE, ERROR and QUEUED) don't make sense.
```

File: apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/pac/McsClaimResponseTransformerV2.java [hunk 2]
```diff
@@ -132,21 +141,37 @@ private static ClaimResponse.ClaimResponseStatus getStatus(RdaMcsClaim claimGrou
...
+   * @param claimGroup The {@link RdaMcsClaim} object to get the status from.
+   * @return The {@link ClaimResponse.RemittanceOutcome} associated with the given data's status
+   *     code.
+   */
   private static ClaimResponse.RemittanceOutcome getOutcome(RdaMcsClaim claimGroup) {
     ClaimResponse.RemittanceOutcome outcome;
 
     if (claimGroup.getIdrStatusCode() == null) {
-      outcome = ClaimResponse.RemittanceOutcome.QUEUED;
+      outcome = ClaimResponse.RemittanceOutcome.PARTIAL;
```
Comment (ericbuckley) [inline apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/pac/McsClaimResponseTransformerV2.java:156]: I think we may need to revert this change.  For both FISS and MCS, I'd like the empty string or NULL values to map to QUEUED.  Its the other random characters that we might see, like `&` or `+`, actually character values that aren't defined in our mappings, should be mapped to PARTIAL.  I'll update the confluence page with this.

**Review by ericbuckley — APPROVED** [review]

**Review by mjburling — COMMENTED** [review]
Neat! I think this looks just fine. I was only curious if this needed any updates to a data dictionary and if so, have they already been made?

**Comment by ericbuckley** [comment]
> Neat! I think this looks just fine. I was only curious if this needed any updates to a data dictionary and if so, have they already been made?

@mjburling Not at the moment.  Right now, our data SMEs are keeping a separate data dictionary for all the PACA data (Claim and ClaimResponse FHIR resources) than what the BFD team manages.  These changes are being added into that DD this sprint (PACA-525).  In the future, as part of the transition plan, PACA has a TODO item to merge these two processes so that the BFD team knows exactly what to do when any of the 5 FHIR elements change.

**Review by christrotteradhoc — APPROVED** [review]

**Review by mackec1 — APPROVED** [review]

**Pull request merged**